desc/equilibrium/coords.py

```diff
@@ -111,19 +111,29 @@ def periodic(x):
 
     # do surface average to get iota once
     if "iota" in profiles and profiles["iota"] is None:
-        profiles["iota"] = eq.get_profile("iota", params=params)
+        profiles["iota"] = (
+            kwargs.pop("iota")
+            if "iota" in kwargs
+            else eq.get_profile("iota", params=params)
+        )
         params["i_l"] = profiles["iota"].params
+    else:
+        kwargs.pop("iota", None)
 
     @functools.partial(jit, static_argnums=1)
     def compute(y, basis):
         grid = Grid(y, sort=False, jitable=True)
         data = {}
         if "iota" in deps:
-            data["iota"] = profiles["iota"](grid, params=params["i_l"])
+            data["iota"] = profiles["iota"](grid, params=params["i_l"], jitable=True)
```

Comment:
i think you can change this to `profiles['iota'].compute` here to avoid the jitable issues